src/uct/rocm/copy/rocm_copy_ep.c

```diff
@@ -40,23 +45,62 @@ UCS_CLASS_DEFINE_DELETE_FUNC(uct_rocm_copy_ep_t, uct_ep_t);
      ucs_trace_data(_fmt " to %"PRIx64"(%+ld)", ## __VA_ARGS__, (_remote_addr), \
                     (_rkey))
 
-static UCS_F_ALWAYS_INLINE ucs_status_t
-uct_rocm_copy_ep_zcopy(uct_ep_h tl_ep,
-                                   uint64_t remote_addr,
-                                   const uct_iov_t *iov,
-                                   int is_put)
+ucs_status_t uct_rocm_copy_ep_zcopy(uct_ep_h tl_ep,
+                                    uint64_t remote_addr,
+                                    const uct_iov_t *iov,
+                                    uct_rkey_t rkey,
+                                    int is_put)
 {
-    size_t size = uct_iov_get_length(iov);
+    hsa_status_t status;
+    hsa_agent_t agent;
+    void *src_addr, *dst_addr;
+    void *host_ptr, *dev_ptr, *mapped_ptr;
+    size_t offset;
 
-    if (!size) {
-        return UCS_OK;
+    ucs_trace("remote addr %p rkey %p", (void*)remote_addr, (void*)rkey);
+
+    size_t size = uct_iov_get_length(iov);
+    uct_rocm_copy_iface_t *iface = ucs_derived_of(tl_ep->iface, uct_rocm_copy_iface_t);
+    hsa_signal_t signal = iface->hsa_signal;
+    uct_rocm_copy_key_t *rocm_copy_key = (uct_rocm_copy_key_t *) rkey;
```

Comment:
```make[4]: Entering directory '/auto/rdmzsysgwork/sergeyo/ucx/src/uct/rocm'
  CC       copy/libuct_rocm_la-rocm_copy_ep.lo
copy/rocm_copy_ep.c: In function ‘uct_rocm_copy_ep_zcopy’:
copy/rocm_copy_ep.c:62:5: error: ISO C90 forbids mixed declarations and code [-Werror=declaration-after-statement]
   62 |     size_t size = uct_iov_get_length(iov);
      |     ^~~~~~
cc1: all warnings being treated as errors
```

Reply:
@hoopoepg Which compiler / configuration options caused this warning?

Reply:
gcc 9.3 on ubuntu 20.04. it seems there is default C90 standard enabled by default.
and this construction violates coding guide lines: all variable declarations should be prior to function code (prior to assert)

Reply:
Thanks, fixed.